lib/smf/smf.c

```diff
@@ -38,7 +38,7 @@ static const struct smf_state *get_child_of(const struct smf_state *states,
 {
 	const struct smf_state *state = states;
 
-	while (state) {
+	while (state != NULL) {
```

Comment:
Please squash these 2 commits together.  No sense introducing code that violates a misra rule only to fix it in the next commit.